crates/net/network/src/session/mod.rs

```diff
@@ -201,6 +210,8 @@ impl SessionManager {
             self.hello.clone(),
             self.status,
             self.fork_filter.clone(),
+            self.blacklisted_peers.clone(),
+            self.blacklisted_ips.clone(),
```

Comment:
hmm, I don't really like this here.

instead of checking the inclusion in the list during authentication, I'd rather do this when the pending session task resolves.

Also imo the right place for this check would be the `PeerManager` which is notified for new incoming connections.

Reply:
We should add some guidelines on what should go where, eg this peer manager recommendation was not obvious to me